src/gsy_e/models/market/future.py

```diff
@@ -54,9 +88,46 @@ def __init__(self, bc: Optional[NonBlockchainInterface] = None,
                          readonly=readonly, grid_fee_type=grid_fee_type,
                          grid_fees=grid_fees, name=name, in_sim_duration=True)
 
-        self.slot_bid_mapping: Dict[DateTime, List[Bid]] = {}
-        self.slot_offer_mapping: Dict[DateTime, List[Offer]] = {}
-        self.slot_trade_mapping: Dict[DateTime, List[Trade]] = {}
+        self._offers = FutureOrders()
+        self._bids = FutureOrders()
+
+    @property
+    def offers(self) -> FutureOrders:
+        """Return the {offer_id: offer} mapping."""
+        return self._offers
+
+    @offers.setter
+    def offers(self, orders) -> None:
+        """Wrap the setter of _orders in order to build a FutureOrders object."""
+        self._offers = FutureOrders(orders)
+
+    @property
+    def bids(self) -> FutureOrders:
+        """Return the {bid_id: bid} mapping."""
+        return self._bids
+
+    @bids.setter
+    def bids(self, orders) -> None:
+        """Wrap the setter of _orders in order to build a FutureOrders object."""
+        self._bids = FutureOrders(orders)
+
+    @property
+    def slot_bid_mapping(self):
+        """Return the {time_slot: [bids_list]} mapping."""
+        return self.bids.slot_order_mapping
+
+    @property
+    def slot_offer_mapping(self) -> Dict:
+        """Return the {time_slot: [offers_list]} mapping."""
+        return self.offers.slot_order_mapping
+
+    @property
+    def slot_trade_mapping(self) -> Dict:
+        """Return the {time_slot: [trades_list]} mapping."""
+        mapping = {time_slot: [] for time_slot in self.slot_bid_mapping.keys()}
+        for trade in self.trades:
+            mapping[trade.time_slot].append(trade)
+        return mapping
```

Comment:
The is not very efficient. So only the lookup in this property will take a lot of computational effort. Why not keeping the `slot_trade_mapping` as it was before because there are not two buffers that need to be synced?

Reply:
@hannesdiedrich IMO there is a decision to be made here
First of all, the trades is a list and not a dictionary, so the solution introduced in this PR isn't relevant in the trades case.
To fix this, I can implement another class which acts like a list (and keeps track of the slot_trade_mapping).
I wouldn't suggest keeping it as it was because we'll have to pay attention on adding the trade to the list and to the slot_trade_mapping every time (and removing them in other cases).

I decided to go with this property solution because the slot_trade_mapping is only accessed once, when exporting the data (significantly less times than adding new trade to the list), so imo it's not much computation

the slightly more computation needed is balanced with the less memory used to store the mapping, both are minors

Reply:
Ok, You are probably right, if we only need the `self.slot_trade_mapping` only for the export now, leave it as is. We can add more logic later, if we need to access this property inside the market too.